feat: add grafana and prometheus to stack

cjorge-graphops opened the pull request:
- adds grafana and prometheus to the stack, configured to scrape clickhouse and redpanda (working on my setup)
- remove an unused config in settings.xml, and use it (that pre-existing confing was getting rejected by clickhouse)
- add a dashboard for key clickhouse / redpanda metrics

cjorge-graphops committed:
feat: add grafana and prometheus to stack

cjorge-graphops committed:
feat: add dashboard

juanmardefago committed:
fix: export -> expose

juanmardefago reviewed (commented):
After fixing the export typo, the docker compose works, and grafana goes up well, but the grafana dashboard is showing errors with the metrics, so I'm assuming there might be either issues with the metrics themselves, or with how grafana uses them.

Example:
![image](https://github.com/user-attachments/assets/a560371d-8189-40e2-97a1-1756f2b5b96c)

Unsure if this was expected (and the PR was essentially to set up the bases for those metrics later down the road), or if those metrics should be working 😅 

cc: @cjorge-graphops

cjorge-graphops committed:
fix: metrics buckets time-window length

cjorge-graphops commented:
> After fixing the export typo, the docker compose works, and grafana goes up well, but the grafana dashboard is showing errors with the metrics, so I'm assuming there might be either issues with the metrics themselves, or with how grafana uses them.
> 
> Example: ![image](https://private-user-images.githubusercontent.com/17829741/427635682-a560371d-8189-40e2-97a1-1756f2b5b96c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNTkzNDEsIm5iZiI6MTc0MzE1OTA0MSwicGF0aCI6Ii8xNzgyOTc0MS80Mjc2MzU2ODItYTU2MDM3MWQtODE4OS00MGUyLTk3YTEtMTc1NmYyYjViOTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI4VDEwNTA0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwNzBkYzAyYTBjYTdmNTgwNzhjYmQzMjBlMmY5YmFkYjVjMDljNDM4YmEzNjIxZmM1ZDc5NzczNWZlMjU1ZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uzml4Cj6g7wJVpgmO3Tcr9t5Cg_tN45zFzBkDLwxAk0)
> 
> Unsure if this was expected (and the PR was essentially to set up the bases for those metrics later down the road), or if those metrics should be working 😅
> 
> cc: @cjorge-graphops

Ty @juanmardefago , try now. This: https://github.com/graphops/qos-oracle-v2/pull/14/commits/d058acfe6788bc14c4317308bc2c257783be300e ought to get them all displaying. I'll iterate more on the dashboard after I do the notes

cjorge-graphops committed:
feat: refine dashboard

feat: add PARTITION BY, to partition data daily. Partitions will
optimize data management when aligned with TTL boundaries, so that
entire partitions can be dropped, or queries can be optimized by
ignoring irrelevant whole partitions.

juanmardefago reviewed (approved):
Works now! Looking good!
